demos/common/cpp/utils/include/utils/shared_tensor_allocator.hpp

```diff
@@ -19,7 +19,7 @@
 #include <opencv2/core.hpp>
 #include <openvino/runtime/allocator.hpp>
 
-class SharedTensorAllocator : public ov::AllocatorImpl {
+class SharedTensorAllocator final : public ov::AllocatorImpl {
```

Comment:
How does `final` solves the problem of nonvirtual destructor?

Reply:
it removes warning without modification of the openvino source)

Reply:
I understand, but why finale helps here?

Reply:
In case if there is a memory leak - it doesn't. It's just a temporarily solution.

Reply:
Mark it with TODO, please. Will ov add a virtual destructor?

Reply:
Idk, I haven't prepared the PR with fix yet.

Reply:
False positive warning https://github.com/openvinotoolkit/openvino/pull/11092. So let's leave `final` specifier to prevent it's emerging.

Reply:
You should suppress the warning using something like `#pragma`

Reply:
Warning pops up in `new_allocator.h` file from from stl:
```
/usr/bin/../lib/gcc/x86_64-linux-gnu/9/../../../../include/c++/9/ext/new_allocator.h:152:4: warning: destructor called on non-final 'SharedTensorAllocator' that has virtual functions but non-virtual destructor [-Wdelete-non-abstract-non-virtual-dtor]
        { __p->~_Up(); }
```
So when I tried to supress it in our code with
```
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wdelete-non-abstract-non-virtual-dtor"
    auto allocator = std::make_shared<SharedTensorAllocator>(mat);
    #pragma clang diagnostic pop
```
It has no effect.  Warning emerges from [here](https://github.com/openvinotoolkit/open_model_zoo/blob/98d967b057dadb760ff4e292cddadd2f26d8c99f/demos/common/cpp/utils/include/utils/ocv_common.hpp#L99):
```
/home/ivikhrev/work/open_model_zoo/demos/common/cpp/utils/include/utils/ocv_common.hpp:102:27: note: in instantiation of function template specialization 'std::make_shared<SharedTensorAllocator, const cv::Mat &>' requested here
    auto allocator = std::make_shared<SharedTensorAllocator>(mat);
```
Any suggestions?

Reply:
Try wrapping `SharedTensorAllocator` definition

Reply:
What does the pragma suppress then?) Are you able to reproduce the warning and suppress it on smaller example?

Reply:
Other warnings supressed successfully, I tested with unused var in code, wrapped it  with:

```
    #pragma clang diagnostic push
    #pragma clang diagnostic ignored "-Wunused-variable"
    int a;
    #pragma clang diagnostic pop
```

Reply:
You can try suppressing in in the ov header, just to learn the way. If it doesn't work, is it clang bug?

Reply:
> What does the pragma suppress then?)

I think it isn't supressed because warning pops up in stl, so suppressing also should be there.

Reply:
Try putting it there)

Reply:
> Try putting it there)

Supressing in `usr/include/c++/9/ext/new_allocator.h:152:4:` works fine:) Seems to supress this warning in our code we should either use `final` or pass flag `-Wdelete-non-abstract-non-virtual-dtor`   to compiler to supress all warnings of this kind

Reply:
Ok. I think keeping `final` with explaining comment is better

Reply:
done